### DISCUSSION TITLE
Why do some files end with .server.ts or .client.ts -- is this convention, or is there semantic meaning?

### DISCUSSION THREAD

**Discussion by vedantroy** [remix-run/remix#2689] [Q&A] [answered]
I notice some repos have files that end with `.server.ts`, for example: `auth.server.ts`. Does remix treat files that end with ".server.ts" / ".server.tsx" differently (besides `entry.server.tsx`), or is this just convention?

**Comment by kiliman** [comment, marked as answer]
This convention explicitly tells the Remix compiler to only bundle this file for the "server" or the "client". Remix is typically smart about which code belongs where, but some packages are not free of side-effects, which make tree-shaking problematic.

You can read more about it at https://remix.run/docs/en/v1/guides/constraints

**Reply by machour** [reply]
The `.client.ts` suffix doesn't seem to be documented on that guide entry tho

**Reply by phil-halo** [reply]
Agree - this should be more explicit in the docs.

**Reply by kiliman** [reply]
Ryan is revamping the docs for both Remix and React Router as they share a lot of API.

**Comment by sys13** [comment]
It's in here now: https://remix.run/docs/en/main/discussion/server-vs-client
